packages/swap/src/routes/swap.ts

```diff
@@ -182,10 +198,26 @@ router.post(
       throw ServiceError.badRequest('invalid request body');
     }
 
-    const { sourceChainExpiryBlock, ...response } =
-      await openSwapDepositChannel(result.data);
-
-    res.json(response);
+    const [{ sourceChainExpiryBlock, ...response }, chainInfo] =
+      await Promise.all([
+        openSwapDepositChannel(result.data),
+        prisma.chainTracking.findFirst({
+          where: {
+            chain: result.data.srcChain,
+          },
+        }),
+      ]);
+
+    const depositChannelExpiryTime = calculateExpiryTime({
+      chain: result.data.srcChain,
+      startBlock: chainInfo?.height,
+      expiryBlock: sourceChainExpiryBlock,
+    });
+
+    res.json({
+      ...response,
```

Comment:
don't want to lose this
```suggestion
      sourceChainExpiryBlock,
      ...response,
```